### ISSUE TITLE
Usage of `S` could be more clear

### ISSUE THREAD

**Issue by ericlagergren** [emanjon/draft-mattsson-cfrg-aes-gcm-sst#9]
It's not bad, but I had to double check that I was reading it right.

An example of what might be more clear:

```
5. Let S = zeropad(A) || zeropad(ct)
6. Let L = LE64(len(ct)) || LE64(len(A))
7. Let X = POLYVAL(H, S[0], S[1], ..., S[num_chunks(S)-1])
8. Let full_tag = POLYVAL(Q, X XOR L) XOR M
```

**Comment by emanjon** [comment]
Thanks. So basically this is two suggestions:
- Separate S and L
- Use num_chunks() instead of m+n

The changes would apply to both encryption and decryption

Current specification is

```
5. Let S = zeropad(A) \|\| zeropad(ct) \|\| LE64(len(ct)) \|\| LE64(len(A))
6. Let X = POLYVAL(H, S[0], S[1], ..., S[m + n - 1])
7. Let full_tag = POLYVAL(Q, X XOR S[m + n]) XOR M
```

**Comment by ericlagergren** [comment]
Yeah. Splitting up S into S and L more clearly describes your intent (two different POLYVAL inputs).

I think `num_chunks` is a more explicit way of saying "use every chunk in S," but there might be a better way of expressing that. Maybe `POLYVAL(H, S...)`? Not sure.

**Comment by emanjon** [comment]
In RFC 8452 they use
`We note that POLYVAL(H, X_1, X_2, ...) is equal to`
That would be 
`Let X = POLYVAL(H, S[0], S[1], ...)`

**Comment by ericlagergren** [comment]
> That would be
> Let X = POLYVAL(H, S[0], S[1], ...)

I like that notation. It's pretty obvious what it means (in my opinion, anyway). And then you don't need to worry about `n` and `m`.

**Comment by emanjon** [comment]
I changed to "Let X = POLYVAL(H, S[0], S[1], ...)" in the PR #10 
Will wait for review from co-authors before merging.

**Comment by emanjon** [comment]
Merged #10 which closes this issue
